EMQX Agent Integration (#2175)
DataDog/integrations-extras: emqx/README.md

# Agent Check: EMQX

## Overview

> This check monitors [EMQX][1] through the Datadog Agent in order to:

Reviewer: Can you please provide a bit more context on:
- What EMQX is/what it does
- The type of data this integration sends into Datadog/why it's valuable for the user

- Collect metrics from EMQX nodes.
- Visualize EMQX performance on the provided dashboard.

## Setup

### Installation

First, [download and launch the Datadog Agent][8].

> Then, manually install the TiDB check. [Instructions vary depending on the environment][10].

Reviewer: ```suggestion
Then, manually install the EMQX check. [Instructions vary depending on the environment][10].
```
Is this the correct link?

> Run `datadog-agent integration install -t datadog-emqx==<INTEGRATION_VERSION>`.

Reviewer: ```suggestion
Run `datadog-agent integration install -t datadog-emqx==1.0.0`.
```

### Configuration

> 1. Edit the `emqx/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your HikariCP performance data.

Reviewer: Typo?
```suggestion
1. Edit the `emqx/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your EMQX performance data.
```

# Agent Check: EMQX

## Overview

[EMQX][1] is a highly scalable, open-source MQTT broker designed for IoT (Internet of Things). MQTT stands for Message Queuing Telemetry Transport, which is a lightweight, publish-subscribe network protocol that transports messages between devices.

**Key features of EMQX:**
- Scalability: EMQX can handle millions of concurrent MQTT connections, making it suitable for IoT applications that require handling a large number of devices.
- Reliability: It provides stable and reliable message delivery, ensuring that data is successfully transferred between devices and servers.
- Low latency: Designed for scenarios requiring low-latency communication.
- High throughput: Capable of processing a high volume of messages efficiently.
- Clustering: EMQX can be deployed in a distributed cluster to enhance performance and reliability.

> Integrating EMQX with Datadog is valuable for several reasons:

Reviewer: ```suggestion
```
Don't think we need this line

**Types of data sent to Datadog:**
- Metrics: This includes performance metrics like message throughput (messages sent/received per second), number of connected clients and more.

- Node performance: Monitoring individual node performance in a cluster, such as latency, load, and operational metrics.

- Operational health: Data about the health of the MQTT broker, including, error rates, and other critical indicators.

> The integration of EMQX with Datadog enriches monitoring capabilities, providing valuable insights into the performance and health of MQTT brokers. This is especially beneficial in IoT applications where efficient, reliable, and real-time data transmission is critical.

Reviewer: Can we move this line above so it's above "Types of data sent to Datadog"?

## Setup

### Installation

> First, [download and launch the Datadog Agent][2].

Reviewer: This step isn't needed since all Datadog users should have the agent installed
```suggestion
```

> Then, manually install the EMQX check. [Instructions vary depending on the environment][3].

Reviewer: ```suggestion
Manually install the EMQX check (note that [instructions may change based on your environment][2]):
```

# Agent Check: EMQX

## Overview

> [EMQX][1] is a highly scalable, open-source MQTT broker designed for IoT (Internet of Things) . MQTT stands for Message Queuing Telemetry Transport, which is a lightweight, publish-subscribe network protocol that transports messages between devices.

Reviewer 2: ```suggestion
[EMQX][1] is a highly scalable, open-source MQTT broker designed for IoT (Internet of Things). MQTT stands for Message Queuing Telemetry Transport, which is a lightweight, publish-subscribe network protocol that transports messages between devices.
```

> **Key Features of EMQX**:

Reviewer 2: ```suggestion
**Key features of EMQX:**
```

- Scalability: EMQX can handle millions of concurrent MQTT connections, making it suitable for IoT applications that require handling a large number of devices.
- Reliability: It provides stable and reliable message delivery, ensuring that data is successfully transferred between devices and servers.
- Low Latency: Designed for scenarios requiring low-latency communication.

> - High Throughput: Capable of processing a high volume of messages efficiently.

Reviewer 2: ```suggestion
- High throughput: Capable of processing a high volume of messages efficiently.
```

- Clustering: EMQX can be deployed in a distributed cluster to enhance performance and reliability.

> Integrating EMQX with Datadog is valuable for several reasons:

Reviewer 2: I recommend deleting this line or moving it under "Why it's valuable", it doesn't add meaning and seems to have been left here when other content was moved or rewritten.

> Types of Data Sent to Datadog:

Reviewer 2: ```suggestion
**Types of data sent to Datadog:**
```

- Metrics: This includes performance metrics like message throughput (messages sent/received per second), number of connected clients and more.

> - Node Performance: Monitoring individual node performance in a cluster, such as latency, load, and operational metrics.

Reviewer 2: ```suggestion
- Node performance: Monitoring individual node performance in a cluster, such as latency, load, and operational metrics.
```

> - Operational Health: Data about the health of the MQTT broker, including, error rates, and other critical indicators.

Reviewer 2: ```suggestion
- Operational health: Data about the health of the MQTT broker, including, error rates, and other critical indicators.
```

> **Why It's Valuable**:

Reviewer 2: ```suggestion
**Why it's valuable:**
```

> - Dashboard Visualization: Datadog provides a visual dashboard that helps in understanding the data and metrics at a glance, which is easier than interpreting raw data.

Reviewer 2: ```suggestion
- Dashboard visualization: Datadog provides a visual dashboard that helps in understanding the data and metrics at a glance, which is easier than interpreting raw data.
```

>   Scalability Insights: Understanding how the system performs under different loads can guide decisions on scaling the infrastructure.

Reviewer 2: ```suggestion
  Scalability insights: Understanding how the system performs under different loads can guide decisions on scaling the infrastructure.
```

- Troubleshooting: Identifying patterns and anomalies in the data can help in diagnosing and resolving issues quickly.

> - Customized Alerts: Setting up alerts based on specific metrics or thresholds can help in proactively managing the system.

Reviewer 2: ```suggestion
- Customized alerts: Setting up alerts based on specific metrics or thresholds can help in proactively managing the system.
```

> The integration of EMQX with Datadog enriches the monitoring capabilities, providing valuable insights into the performance and health of MQTT brokers. This is especially beneficial in IoT applications where efficient, reliable, and real-time data transmission is critical.

Reviewer 2: ```suggestion
The integration of EMQX with Datadog enriches monitoring capabilities, providing valuable insights into the performance and health of MQTT brokers. This is especially beneficial in IoT applications where efficient, reliable, and real-time data transmission is critical.
```

## Setup

### Installation

First, [download and launch the Datadog Agent][2].

Then, manually install the EMQX check. [Instructions vary depending on the environment][3].

Run `datadog-agent integration install -t datadog-emqx==1.0.0`.

### Configuration

> 1. Edit the `emqx/conf.yaml` file, in the `conf.d/` folder at the root of your Agent's configuration directory to start collecting your EMQX performance data.

Reviewer 2: ```suggestion
1. Edit the `emqx/conf.yaml` file, located in the `conf.d/` folder at the root of your Agent's configuration directory, to start collecting your EMQX performance data.
```

   
2. [Restart the Agent][4].

### Validation

[Run the Agent's status subcommand][5] and look for `emqx` under the Checks section.

## Data Collected

### Metrics

See [metadata.csv][6] for a list of metrics provided by this integration.

### Service Checks

See [service_checks.json][7] for a list of service checks provided by this integration.

### Events

EMQX does not include any events.

## Troubleshooting

> Need help? Contact [Datadog support][8].

Reviewer 3: This should be the partner support team
